### ISSUE TITLE
Datastore emulator doesn't start

### ISSUE THREAD

**Issue by fedoseev-vitaliy** [GoogleCloudPlatform/cloud-sdk-docker#144]
I'm unable to start datastore emulator after updating to latest version 212.

`
ERROR: (gcloud.beta.emulators.datastore.start) Unable to prepare the data directory for the emualtor
Executing: /usr/lib/google-cloud-sdk/platform/cloud-datastore-emulator/cloud_datastore_emulator create --project_id=project /root/.config/gcloud/emulators/datastore
[datastore] Exception in thread "main" java.lang.UnsupportedClassVersionError: com/google/cloud/datastore/emulator/CloudDatastore : Unsupported major.minor version 52.0
[datastore] 	at java.lang.ClassLoader.defineClass1(Native Method)
[datastore] 	at java.lang.ClassLoader.defineClass(ClassLoader.java:808)
[datastore] 	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
[datastore] 	at java.net.URLClassLoader.defineClass(URLClassLoader.java:442)
[datastore] 	at java.net.URLClassLoader.access$100(URLClassLoader.java:64)
[datastore] 	at java.net.URLClassLoader$1.run(URLClassLoader.java:354)
[datastore] 	at java.net.URLClassLoader$1.run(URLClassLoader.java:348)
[datastore] 	at java.security.AccessController.doPrivileged(Native Method)
[datastore] 	at java.net.URLClassLoader.findClass(URLClassLoader.java:347)
[datastore] 	at java.lang.ClassLoader.loadClass(ClassLoader.java:430)
[datastore] 	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:323)
[datastore] 	at java.lang.ClassLoader.loadClass(ClassLoader.java:363)
[datastore] 	at sun.launcher.LauncherHelper.checkAndLoadMain(LauncherHelper.java:482)
`

**Comment by logicaleak** [comment]
Having the same issue, in the last build of this docker image, I think java version is broken. I have just `exec`'ed into the image, and ran `java -version`, it seems to be `1.7`, while the datastore emulator expects `1.8` to run. You could start an image from `google/cloud-sdk` and install the correct version of java yourself, but I find it kinda wrong, so I assume we should jump back to the previous version until a patch is ran ?

**Comment by salrashid123** [comment]
thanks for the report.

Looks like the emulator itself needs `java8` now
- [https://cloud.google.com/datastore/docs/tools/datastore-emulator](https://cloud.google.com/datastore/docs/tools/datastore-emulator) 

but the package installer for this docker image acquires 7.

I'll file an internal bug to update and see if/how it impacts the other parts of gcloud

```
$ docker run -ti  google/cloud-sdk:212.0.0 /bin/bash
root@508d84b73ee5:/# java -version
java version "1.7.0_181"
OpenJDK Runtime Environment (IcedTea 2.6.14) (7u181-2.6.14-1~deb8u1)
OpenJDK 64-Bit Server VM (build 24.181-b01, mixed mode)
```

----
```
# apt-cache depends --recurse --no-recommends --no-suggests --no-conflicts --no-breaks --no-replaces --no-enhances --no-pre-depends google-cloud-sdk-datastore-emulator | grep "^\w" | sort -u | grep jdk 
openjdk-7-jdk
openjdk-7-jre
openjdk-7-jre-headless
```

```
root@4751c6dccaa5:/# gcloud beta emulators datastore start
Executing: /usr/lib/google-cloud-sdk/platform/cloud-datastore-emulator/cloud_datastore_emulator create --project_id=foo /root/.config/gcloud/emulators/datastore
[datastore] Exception in thread "main" java.lang.UnsupportedClassVersionError: com/google/cloud/datastore/emulator/CloudDatastore : Unsupported major.minor version 52.0
[datastore] 	at java.lang.ClassLoader.defineClass1(Native Method)
[datastore] 	at java.lang.ClassLoader.defineClass(ClassLoader.java:808)
```

**Comment by wsh** [comment]
We've fixed this internally; barring release issues, we expect the fix to arrive in version 214.0.0. Sorry for the inconvenience!

**Comment by hiroshi** [comment]
Hi, with google/cloud-sdk:214.0.0, I got another error.
```
$ docker run --rm -ti google/cloud-sdk:214.0.0 gcloud beta emulators datastore start
ERROR: (gcloud.beta.emulators.datastore.start) The java executable on your PATH is not a Java 8+ JRE. The Google Cloud Datastore emulator requires a Java 8+ JRE installed and on your system PATH
```

**Comment by wsh** [comment]
Hi there @hiroshi: I'm so sorry, the fix was rolled back (release issue) and this bug should have been updated to reflect that. We're fixing the fix and I'll report back when we know it sticks.

**Comment by wsh** [comment]
This should be fixed in 215.0.0, available today. Thanks for your patience, everyone!

**Comment by salrashid123** [comment]
`215.0.0` is out now w/ the emulator package updated.  I've pushed the release and got the images on dockerhub built.

```
$ docker run -ti google/cloud-sdk:215.0.0 /bin/bash
root@f58088ab9709:/# gcloud config set project blah
Updated property [core/project].
root@f58088ab9709:/# gcloud beta emulators datastore start
Executing: /usr/lib/google-cloud-sdk/platform/cloud-datastore-emulator/cloud_datastore_emulator create --project_id=blah /root/.config/gcloud/emulators/datastore
[datastore] Sep 05, 2018 7:38:43 PM com.google.cloud.datastore.emulator.CloudDatastore$CreateAction$1 apply
[datastore] INFO: Provided project_id to Cloud Datastore emulator create command, which is no longer necessary.
[datastore] Created new Cloud Datastore project in '/root/.config/gcloud/emulators/datastore'.
```

---

Some notes since this is more than just a version update:

- `:latest`, ':slim`, `:<version>` images
   -  upgrade to debian  `jessie` -> `stretch`   (the former doens't have jdk8 w/o backports)
   -  add `gnupg` (seems its needed on `stretch` if i add an apt source..); 

- `:alpine`: i also added it to the alpine image for consistency but after the push, had second thoughts:
    its not necessary and adds a bit more MB;  I'll remove the extra `gnupg` in `:216.0.0` release; my bad

```
    REPOSITORY                     TAG                 IMAGE ID            CREATED             SIZE
    google/cloud-sdk               215.0.0-alpine      ae82c594dfc7        2 minutes ago       208MB
    google/cloud-sdk               214.0.0-alpine      b210196913dc        8 days ago          194MB
```

- `:legacy` image uses `FROM gcr.io/google_appengine/base`
  and that is still `jessie`, i'm hesitating on enabling backports to get jdk8 and doubt anyone is really using this still.
  I'm leaving it as-is.  We'll delete this image anyway soon after announce on the README

    for ref: this'd add in the jdk8 bits for `jessie` for the GAE based image

    ```dockerfile
     RUN echo "deb http://http.debian.net/debian jessie-backports main" > /etc/apt/sources.list.d/google-cloud-sdk.list && \
         apt-get update && \
         apt-get install -t jessie-backports -y openjdk-8-jre-headless
     RUN apt-get install -y -qq --no-install-recommends wget unzip python php5-mysql php5-cli php5-cgi openjdk-8-jre-headless openssh-client python-openssl && apt-get clean
    ```
